File: Assignment 2/leapyear.cpp
#include<iostream>
using namespace std;
class Year
{
    int num;
    public:void leapyear()
    {
        cout<<"Enter the year to be checked:\n";
        cin>>num;
        if (((num % 4 == 0) && (num % 100 != 0)) || (num % 400 == 0))
        cout<<num<<" is a leap year";
        else
        cout<<num<<" is not a leap year";
    }
};
int main()
{
    Year check;
    check.leapyear();
}

File: JAVA/Assignment2/LeapYear.java
import java.util.Scanner;
public class LeapYear
{
   public static void main(String args[])
   {
	  int num;
      System.out.println("Enter the year::");
      Scanner sc = new Scanner(System.in);
	  num=sc.nextInt();
	  if (((num % 4 == 0) && (num % 100 != 0)) || (num % 400 == 0))
		 System.out.println("The entered year is a leap year ="+ num); 
	 else
		 System.out.println("The entered is not a leap year ="+num); 
   }
}